site/zenodo_rdm/legacy/serializers/schemas.py

```diff
@@ -543,7 +541,12 @@ def dump_state(self, result, original, **kwargs):
     @post_dump(pass_original=True)
     def dump_prereserve_doi(self, result, original, **kwargs):
         """Dump prereserved DOI information."""
-        provider = original["pids"]["doi"]["provider"]
+        provider = original.get("pids", {}).get("doi", {}).get("provider")
+
+        # New versions don't have pids provider out of the box
+        if not provider:
+            return result
```

Comment:
When we create a new version, "pids.doi" does not exist and it was blowing up.

Reply:
minor/shelve: we might need to override this in our service so that we go through the same `create` steps when creating a new version.